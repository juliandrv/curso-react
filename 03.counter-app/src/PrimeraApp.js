// import React from "react"; //Descomentar si aparece error
// import { Fragment } from "react";
import { PropTypes } from "prop-types";

const PrimeraApp = ({ saludo, subtitulo }) => {
  // const saludo = 'Hola Mundo';
  // const saludo = 123;
  // const saludo = true; // Los valores booleanos no imprimen nada
  // const saludo = [1,2,'Hola',4];
  // const saludo = {
  //   nombre: 'Juli√°n',
  //   edad: 32
  // };


  return (
    <>
      <h1>{saludo}</h1>
      {/* Muestra un objeto formateado */}
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <p>{subtitulo}</p>
    </>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
  subtitulo: 'Mi primera app React'
}

export default PrimeraApp;