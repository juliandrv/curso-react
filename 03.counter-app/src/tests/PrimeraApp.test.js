// import "@testing-library/jest-dom/extend-expect";
// import { render } from "@testing-library/react";
import '../setUpTests'
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe('Pruebas en <PrimeraApp />', () => {
  // test('debe mostrar el saludo Hola, soy Julián', () => {
  //   const saludo = 'Hola, soy Julián';

  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);

  //   expect(getByText(saludo)).toBeInTheDocument();
  // })
  test('debe mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, soy Julián';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();


  });
})
