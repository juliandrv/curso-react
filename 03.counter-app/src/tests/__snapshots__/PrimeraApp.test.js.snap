// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Juli√°n
  </h1>
  <p>
    Mi primera app React
  </p>
</Fragment>
`;
